web/src/user_profile.ts

```diff
@@ -121,10 +122,12 @@ export function get_user_id_if_user_profile_modal_open(): number | undefined {
     return undefined;
 }
 
-export function update_user_profile_streams_list_for_users(user_ids: number[]): void {
+export async function update_user_profile_streams_list_for_users(
+    user_ids: number[],
+): Promise<void> {
     const user_id = get_user_id_if_user_profile_modal_open();
     if (user_id && user_ids.includes(user_id) && user_streams_list_widget !== undefined) {
-        const user_streams = stream_data.get_streams_for_user(user_id).subscribed;
+        const user_streams = (await stream_data.get_streams_for_user(user_id)).subscribed;
```

Comment:
It's probably fine for this to use `await`, but I wonder if this will never actually wait, because `user_streams_list_widget` should only be populated if we've in fact already received the data?

(I think it's fine to just use the `await` approach rather than some other API that asserts we already have it, since it seems non-disruptive)